pkg/render/istio/istio_test.go

```diff
@@ -0,0 +1,855 @@
+			cfg.Installation.KubernetesProvider = operatorv1.ProviderGKE
+			crds, component, err := istio.Istio(cfg)
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(crds).NotTo(BeNil())
+			Expect(component).NotTo(BeNil())
+		})
+	})
+
+	Describe("ResolveImages", func() {
+		It("should resolve all images correctly", func() {
+			_, component, err := istio.Istio(cfg)
+			Expect(err).ShouldNot(HaveOccurred())
+
+			imageSet := &operatorv1.ImageSet{
+				Spec: operatorv1.ImageSetSpec{
+					Images: []operatorv1.Image{
+						{Image: "tigera/istio-pilot", Digest: "sha256:test-pilot-digest"},
+						{Image: "tigera/istio-install-cni", Digest: "sha256:test-cni-digest"},
+						{Image: "tigera/istio-ztunnel", Digest: "sha256:test-ztunnel-digest"},
+						{Image: "tigera/istio-proxyv2", Digest: "sha256:test-proxyv2-digest"},
+					},
+				},
+			}
+
+			err = component.ResolveImages(imageSet)
+			Expect(err).ShouldNot(HaveOccurred())
+		})
+
+		It("should fail when required images are missing", func() {
+			_, component, err := istio.Istio(cfg)
+			Expect(err).ShouldNot(HaveOccurred())
+
+			imageSet := &operatorv1.ImageSet{
+				Spec: operatorv1.ImageSetSpec{
+					Images: []operatorv1.Image{},
+				},
+			}
+
+			err = component.ResolveImages(imageSet)
+			Expect(err).Should(HaveOccurred())
+		})
+	})
+
+	Describe("Objects", func() {
+		var (
+			crds      *istio.IstioComponentCRDs
+			component *istio.IstioComponent
+			err       error
+		)
+
+		BeforeEach(func() {
+			crds, component, err = istio.Istio(cfg)
+			Expect(err).ShouldNot(HaveOccurred())
+
+			imageSet := &operatorv1.ImageSet{
+				Spec: operatorv1.ImageSetSpec{
+					Images: []operatorv1.Image{
+						{Image: "tigera/istio-pilot", Digest: "sha256:test-pilot-digest"},
+						{Image: "tigera/istio-install-cni", Digest: "sha256:test-cni-digest"},
+						{Image: "tigera/istio-ztunnel", Digest: "sha256:test-ztunnel-digest"},
+						{Image: "tigera/istio-proxyv2", Digest: "sha256:test-proxyv2-digest"},
+					},
+				},
+			}
+			err = component.ResolveImages(imageSet)
+			Expect(err).ShouldNot(HaveOccurred())
+		})
+
+		It("should return CRDs", func() {
+			objsToCreate, objsToDelete := crds.Objects()
+			Expect(objsToCreate).NotTo(BeEmpty())
+			Expect(objsToDelete).To(BeEmpty())
+		})
+
+		It("should return objects to create", func() {
+			objsToCreate, objsToDelete := component.Objects()
+			Expect(objsToCreate).NotTo(BeEmpty())
+			Expect(objsToDelete).To(BeEmpty())
+		})
+
+		It("should include network policies", func() {
+			objsToCreate, _ := component.Objects()
+
+			// Verify istiod policy
+			istiodPolicy, err := rtest.GetResourceOfType[*v3.NetworkPolicy](objsToCreate, istio.IstioIstiodPolicyName, istio.IstioNamespace)
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(istiodPolicy.Spec.Tier).To(Equal(networkpolicy.TigeraComponentTierName))
+			Expect(istiodPolicy.Spec.Selector).To(Equal(networkpolicy.KubernetesAppSelector(istio.IstioIstiodDeploymentName)))
```

Comment:
We typically assert on the contents of the policies more broadly - you've got a couple of fields here but we should be asserting on the ingress / egress types and rules as well.